src/parser.cpp

```diff
@@ -542,11 +546,11 @@ namespace ada::parser {
             input_position = input_size + 1;
           }
           if constexpr (result_type_is_ada_url) {
-            if(!helpers::parse_prepared_path(view, url.type, url.path)) { return url; }
+            helpers::parse_prepared_path(view, url.type, url.path);
```

Comment:
Previously we were returning on the failed path in here. Doesn't this breaking *some* tests?

Reply:
I believe our test coverage does not cover this scenario.

Reply:
parse_prepared_path would always return true.

Reply:
There is no scenario where a path parsing could fail.

Theoretically, this could happen... there are invalid inputs, but the spec. insists that we do not halt the processing.

This is related to this discussion...

https://github.com/ada-url/ada/pull/248

So the path `ab%xy.txt` should be rejected, it is invalid. But we are also instructed to not halt the processing. So return false would be wrong.

There is no scenario where path parsing can stop the processing. At least none in our existing code base. And I don't think it is a matter of testing because we do want `https://google.com/ab%xy.txt` to just 'work'.

I think that the intention of the standard is that we accept pretty much everything. Now, as a secondary step, we could *validate* that the path is ok. But that belongs to a distinct PR.